internal/datastore/crdb/crdb.go

```diff
@@ -82,11 +82,25 @@ func newCRDBDatastore(url string, options ...Option) (datastore.Datastore, error
 
 	configurePool(config, poolConfig)
 
-	pool, err := pgxpool.ConnectConfig(context.Background(), poolConfig)
+	initCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	pool, err := pgxpool.ConnectConfig(initCtx, poolConfig)
 	if err != nil {
 		return nil, fmt.Errorf(errUnableToInstantiate, err)
 	}
 
+	var version crdbVersion
+	if err := queryServerVersion(initCtx, pool, &version); err != nil {
```

Comment:
any reason we do the "pass-in-pointer" of this vs just returning a pointer of the version struct?

Reply:
Makes sense. I just find it a bit riskier since you could accidentally pass in `nil`, but since it is internal to CRDB package, its fine